### DISCUSSION TITLE
Animating scatter plot

### DISCUSSION THREAD

**Discussion by arm2arm** [ahuang11/ahlive#41] [General]
Hello, 

Is it possible to animate the pandas DataFrame with [x,y,z] columns, where animation will go from one scatter plot to another.
lets say:
dfplot=df[["x","y","z"]].copy()
how to animate between 2 scatter plots: dfplot[["x","y"]] and dfplot[["x","z"]] ?

Thanks for the nice project.

**Comment by ahuang11** [comment]
Hi, @arm2arm thanks for your interest!

This is the best I've got so far; I am a bit stumped on how to do it with solely an ah.DataFrame, but maybe it just needs a separate join implementation!
```
import ahlive as ah
import pandas as pd

df = pd.DataFrame({'x': [-1, -2, -3, -4], 'y': [2, 3, 4, 5], 'z': [5, -5, -10, -11]})
df = df.set_index('x').T
ah.overlay([ah.Array(col * 2, df[col], group='same', legend=False) for col in df.columns]).render()   
```
![test](https://user-images.githubusercontent.com/15331990/109250389-36f5e980-77af-11eb-8cc6-3ec96734750c.gif)

**Reply by arm2arm** [reply]
this is nice!

```
import ahlive as ah
import pandas as pd

df = pd.DataFrame({'x': dfdb.xgal, 'y': dfdb.ygal, 'z': dfdb.zgal})
df = df.set_index('x').T
ah.overlay([ah.Array(col * 2, df[col], group='same', legend=False,save='1.gif') for col in df.columns]).render()   

```
![1](https://user-images.githubusercontent.com/2283443/109291362-2c783600-7829-11eb-876b-888c602322c1.gif)

**Comment by ahuang11** [comment]
Oh hey, I did have the same idea back then; forgot about it myself and forgot to document it.
```
df = pd.DataFrame({'x': [-1, -2, -3, -4], 'y': [2, 3, 4, 5], 'z': [5, -5, -10, -11]})
df = df.melt('x')
ah_df = ah.DataFrame(df, 'x', 'value', label='variable', legend=False, color='blue').invert()
ah_df.render()
```

**Reply by ahuang11** [reply]
Although, the method needs some tweaking like passing in labels.

**Comment by arm2arm** [comment]
Somehow the second method doesn't work

```
import ahlive as ah

df = pd.DataFrame({'x': dfdb.xgal, 'y': dfdb.ygal, 'z': dfdb.zgal})
df = df.melt('x')
ah_df = ah.DataFrame(df, 'x', 'value', label='variable', legend=False, color='blue',save='2.gif').invert()

ah_df.render()

```

![2](https://user-images.githubusercontent.com/2283443/109291002-ac51d080-7828-11eb-970b-e521dc825189.gif)

**Reply by ahuang11** [reply]
I think you need to specify chart='scatter'
